lib/manageiq_performance/middleware.rb

```diff
@@ -90,15 +90,29 @@ def save_report env, prefix, long_form, short_form
     end
   end
 
-  def self.profile name = nil, &code
-    name ||= caller.first.match(/`(|.*\s)([a-z_\(\)]+)>?'$/)[2].gsub(/[^a-z_]/,'')
+  def self.profile *args, &code
+    options = args.last.is_a?(Hash) ? args.shift : {}
+    name    = args.shift
+    name    ||= caller.first.match(/`(|.*\s)([a-z_\(\)]+)>?'$/)[2].gsub(/[^a-z_]/,'')
+
     env  = {
       Middleware::PERFORMANCE_HEADER => true,
       "REQUEST_PATH"                 => name,
       "HTTP_MIQ_PERF_TIMESTAMP"      => (Time.now.to_f * 1000000).to_i
     }
 
-    ManageIQPerformance::Middleware.new(code).call(env)
+    if options[:config_changes] || options[:in_memory]
+      config_changes = (options[:config_changes] || {}).dup
```

Comment:
Doing a `.dup` here around them both to avoid overwriting any values when doing the `.merge!`.  Technically we would be duping and empty hash as well, but saves a few characters just to do it this way.  #lazyNVL